client/oidc.go

```diff
@@ -80,9 +121,30 @@ func VerifyPKToken(ctx context.Context, pkt *pktoken.PKToken, provider OpenIdPro
 		return fmt.Errorf("error verifying CIC signature on PK Token: %w", err)
 	}
 
-	if pkt.Cos != nil {
-		if err := pkt.VerifyCosignerSignature(); err != nil {
-			return fmt.Errorf("error verify cosigner signature on PK Token: %w", err)
+	// If no cosigner is set, then we don't require a cosigner.
+	if len(v.AllowedCosigners) != 0 {
+		if pkt.Cos == nil {
+			return fmt.Errorf("cosigner required but no cosigner signature (COS) set in PK Token")
+		}
+
+		var cos *CosignerProvider
+		if cosIss, ok := pkt.Cos.ProtectedHeaders().Get("iss"); !ok {
+			return fmt.Errorf("no COS issuer set in the PK Token")
+		} else {
+			for _, allowedCos := range v.AllowedCosigners {
+				if allowedCos.GetIssuer() == cosIss {
+					cos = &allowedCos
+					break
+				}
+			}
+			if cos == nil {
+				return fmt.Errorf("the COS issuer (%s) in the PK Token is not an list of allowed issuers", cosIss)
+			}
+		}
+		if pkt.Cos != nil {
+			if err := pkt.VerifyCosignerSignature(); err != nil {
+				return fmt.Errorf("error verify cosigner signature on PK Token: %w", err)
+			}
```

Comment:
`if pkt.Cos != nil {` shouldn't be there. It is still secure because on line 126 I check `if pkt.Cos != nil {` and fail if it is nil so the impact here is just a NOP, but it should be removed.

Reply:
Fixed